### PR TITLE
Add show icon in pickup config

### PR DESCRIPTION
**Pull request by UltiNaruto**
Add pickup icon texture in GGuiSys.pak
Fix mapa not updating offsets when adding a pickup icon
Object id generator doesn't consider object IDs such as 0x????F???
Updated pickup icon texture as it was buggy when rotating the map

### CODE REVIEW

File: src/patches.rs [hunk 1]
```diff
@@ -12529,13 +12645,25 @@ fn build_and_run_patches(gc_disc: &mut structs::GcDisc, config: &PatchConfig, ve
...
+                );
+
                 idx = idx + 1;
                 seed = seed + 1;
             }
 
             // Patch extra item locations
             while idx < pickups_config_len {
                 let pickup = pickups[idx].clone(); // TODO: cloning is suboptimal
+                let pickup2 = pickups[idx].clone();
```
Comment (toasterparty) [inline src/patches.rs:12666]: Why is pickup2 needed? Can you maybe give it a better name?

Comment (UltiNaruto) [reply src/patches.rs:12666]: rust would complain of the reference to pickup being already used by modify_pickups_in_mrea and patch_add_item

Comment (UltiNaruto) [reply src/patches.rs:12666]: about the name I don't know what to name it tbh

Comment (toasterparty) [reply src/patches.rs:12666]: just clone it in the param to `modify_pickups_in_mrea` and `patch_add_item` then

Comment (UltiNaruto) [reply src/patches.rs:12666]: What do you mean cloning it in those functions? The reference can't be the same as the one of add_pickups_to_mapa

Try to use only pickup in both modify_pickups_in_mrea and add_pickups_to_mapa and read the error message

**Pull request merged**